internal/exec/stages/files/units.go

```diff
@@ -145,31 +146,41 @@ func parseInstanceUnit(unit types.Unit) (string, string, error) {
 
 // createSystemdPresetFile creates the presetfile for enabled/disabled
 // systemd units.
-func (s *stage) createSystemdPresetFile(presets map[string]*Preset) error {
-	if err := s.relabelPath(filepath.Join(s.DestDir, util.PresetPath)); err != nil {
-		return err
+func (s *stage) createSystemdPresetFiles(presets map[string]*Preset) error {
+
+	//getting directories from presets file for relabling
+	paths := make(map[string]bool)
+	for _, preset := range presets {
+		path := filepath.Dir(s.SystemdPresetPath(preset.scope))
```

Comment:
The `Dir` call and the building of `paths` is to avoid duplicate relabeling, right?  Is that worth the complexity of the extra code?

Reply:
same as https://github.com/coreos/ignition/pull/1303#discussion_r815365077